tests/unit/test_repo.py

```diff
@@ -1,6 +1,53 @@
 import os
 
+import pytest
+
 
 def test_is_dvc_internal(dvc):
     assert dvc.is_dvc_internal(os.path.join("path", "to", ".dvc", "file"))
     assert not dvc.is_dvc_internal(os.path.join("path", "to-non-.dvc", "file"))
+
+
+@pytest.mark.parametrize(
+    "path",
+    [
+        os.path.join("dir", "subdir", "file"),
+        os.path.join("dir", "subdir"),
+        "dir",
+    ],
+)
+def test_find_outs_by_path(tmp_dir, dvc, path):
+    stage, = tmp_dir.dvc_gen({"dir": {"subdir": {"file": "file"}}})
+
+    outs = dvc.find_outs_by_path(path)
+    assert len(outs) == 1
+    assert outs[0].path_info == stage.outs[0].path_info
+
+
+@pytest.mark.parametrize(
+    "path",
+    [
+        os.path.join("dir", "subdir", "file"),
+        os.path.join("dir", "subdir"),
+        "dir",
+    ],
+)
+def test_used_cache(tmp_dir, dvc, path):
+    from dvc.cache import NamedCache
+
+    tmp_dir.dvc_gen({"dir": {"subdir": {"file": "file"}}})
+    expected = NamedCache.make(
+        "local", "d28c9e28591aeb7e303dc6772ffa6f6b.dir", "dir"
+    )
+    expected.add(
+        "local",
+        "8c7dd922ad47494fc02c388e12c00eac",
+        os.path.join("dir", "subdir", "file"),
+    )
+
+    with dvc.state:
+        used_cache = dvc.used_cache([path])
```

Comment:
Filtering is not really tested since there is nothing filtered out. Need to add some file next to `subdir`.